### PR TITLE
Members Only Project

### PR DESCRIPTION
**Pull request by julian3493**
I complete the members-only project with the indications given in the Odin project:

- Use the devise gem for authentication.
- Add a relationship between User and Post models.
- Add restrictions to create posts for people who are not logged.
- Add logic in index view in order to just let see who posts only the members.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
+  end
```
Comment (trekab) [inline app/controllers/posts_controller.rb]: Initialize an empty post this way `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)) This is the preferred way. It also avoids repetition of the `build` method in both the `new` and `create` actions.

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi team! So far so good! :+1: I commend you for implementing flash messages for unsuccessful actions for your project. However, there are some little details for you to work on to finish this project:
- [x] Use a `before_action` to restrict access to the `:new` and `:create` methods (use an ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [x] You are expected to create your custom helper methods in the `ApplicationController` and use these in the above scenario and in the `posts` index view for proper user authentication.
- [x] Check out the comment below for more feedback.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Comment by julian3493** [comment]
> ### Changes Required! ♻️
> Hi team! So far so good! 👍 I commend you for implementing flash messages for unsuccessful actions for your project. However, there are some little details for you to work on to finish this project:
> 
> * [ ]  Use a `before_action` to restrict access to the `:new` and `:create` methods (use an ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> * [ ]  You are expected to create your custom helper methods in the `ApplicationController` and use these in the above scenario and in the `posts` index view for proper user authentication.
> * [ ]  Check out the comment below for more feedback.
> 
> **Kindly make the necessary changes and submit another code review request.**
> _Feel free to reach me on slack for any inquiries about this review._ 😉
> **Treasure Kabareebe**
> [Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (julian3493) [reply app/controllers/posts_controller.rb]: I'm using a before_action to restrict access to some methods including the :new and :create methods, devise give a helper and I'm using this helper.
image

**Review by trekab — APPROVED** [review]
### Approved! :tada: 
Well done, @julian3493! I understand your use of the devise helper methods. You may proceed to the next project.

Happy coding!

**Pull request merged**